Dev

SleeplessShu opened the pull request:
Домашнее задание спринта №10

SleeplessShu committed:
Подключена библиотека Glide

SleeplessShu committed:
Исправлено:
- кнопки перемещения "назад", поделиться и тп. Обработка нажатия перенесена с иконки на Layout

Добавлено:
- recyclerview в activity_search

SleeplessShu committed:
Исправлено:
- цвет статусбара соответствует цвету бэкграунда в каждой активити

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker_ver2/SettingsActivity.kt:
```diff
@@ -18,27 +24,40 @@ class SettingsActivity : AppCompatActivity() {
-        switchTheme.isChecked = isNightModeOn
 
-        switchTheme.setOnCheckedChangeListener { _, isChecked ->
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && AppCompatDelegate.getDefaultNightMode() != AppCompatDelegate.MODE_NIGHT_YES) {
+            val window = window
+            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
+            window.statusBarColor = ContextCompat.getColor(this, R.color.white)
+            window.decorView.systemUiVisibility =
+                window.decorView.systemUiVisibility or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+        }
```
Этот код скопирован в несколько активити, лучше вынести это в отдельную функцию и переиспользовать функцию вместо копирования кода

SleeplessShu replied:
Создана BaseActivity в которую перенесена функция. Теперь она вызывается в каждой активити, кроме основной. 

Если поменять настройки темы, то можно вызывать функцию только для MainActivity, а для остальных брать цвета по умолчанию из темы. Если мне удалось донести мысль, то стоит ли поменять реализацию и повлияет ли это на что-то?

ArturNurtdinov replied:
Реализация норм, но я бы от наследования отказался. Можно просто написать экстеншн функцию и вызывать её

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker_ver2/TrackViewHolder.kt:
```diff
@@ -0,0 +1,32 @@
+            .load(model.artworkUrl100)
+            .placeholder(R.drawable.icon_nointernet)
+            .fitCenter()
+            .transform(RoundedCorners(2))
```
В конструктор `RoundedCorners` нужно передавать кол-во пикселей. Это значит, что предварительно число, передаваемое в конструктор, нужно сконвертировать в пиксели, если мы хотим добиться одинаково скругления на девайсах с разной плотностью экрана

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker_ver2/TrackViewHolder.kt:
```diff
@@ -0,0 +1,32 @@
+    private val artistName: TextView = itemView.findViewById(R.id.compArtistName)
+    private val trackTime: TextView = itemView.findViewById(R.id.compDuration)
+    private val artworkUrl100: ImageView = itemView.findViewById(R.id.compTrackIcon)
+    fun bind(model: TrackData, context: Context) {
```
Можно не прокидывать контекст параметром в функцию, ты можешь взять контекст у любой из вью, например `trackName.context`

ArturNurtdinov commented on app/src/main/res/drawable/icon_nointernet.xml:
Немного не тот ресурс экспортирован из фигмы, нужно вот такой:
![image](https://github.com/SleeplessShu/PlaylistMaker_ver2/assets/49075521/09b15d10-07a4-4cf5-8f47-b30069b90d0d)

SleeplessShu replied:
Это было настолько неочевидно) Я думал, что это просто рисунок альбома и использовал свой)) Исправил

SleeplessShu committed:
Исправлено:
- плэйсхолдер заменен на корректный
- определение цвета статусбара вынесено в функцию, код переиспользуется
- в RoundedCorners добавлен конвертр
- контекст в bind не прокидывается, а берется из view

SleeplessShu committed:
Добавлено:
- рабочий поиск треков
- заглушка при ошибке\отсутствии сети
- заглушка при пустом результате поиска
